src/renderer/component/page/view.jsx

```diff
@@ -49,16 +31,29 @@ class Page extends React.PureComponent<Props, State> {
   }
 
   componentDidMount() {
-    const { children } = this.props;
+    const { loading } = this.props;
+    if (loading) {
+      this.beginLoadingTimeout();
+    }
+  }
 
-    if (!isShowingChildren(children))
-      this.loaderTimeout = setTimeout(() => {
-        this.setState({ showLoader: true });
-      }, LOADER_TIMEOUT);
+  componentDidUpdate(prevProps: Props) {
+    const { loading } = this.props;
+    if (!this.loaderTimeout && !prevProps.loading && loading) {
+      this.beginLoadingTimeout();
+    }
   }
 
   componentWillUnmount() {
-    this.loaderTimeout = null;
+    if (this.loaderTimeout) {
+      this.loaderTimeout = null;
```

Comment:
This will only set the handle to null, you'll want to use `clearTimeout` in some parts of the flow.

Reply:
Whoops